### PR TITLE
Server middleware & login

### PR DESCRIPTION
**Pull request by FunctionalMetatable**

### CODE REVIEW

**Comment by FunctionalMetatable** [comment]
@TheColaber

**Comment by TheColaber** [comment]
oh no vercel wants me to pay 20 dollors, or delete the team

**Comment by FunctionalMetatable** [comment]
> oh no vercel wants me to pay 20 dollors, or delete the team

better to delete the team

**Comment by TheColaber** [comment]
@FunctionalMetatable I'm getting 
image
I don't think you can use the word private in javascript.

**Comment by FunctionalMetatable** [comment]
fixed, hopefully

**Comment by TheColaber** [comment]
Just so you know, I'm reviewing and changing some code. Thanks for keeping this thing active. I've been so busy with my jobs, but I'll try to be more active too.

**Comment by FunctionalMetatable** [comment]
Also, could there be an official Authentication studio?

**Comment by TheColaber** [comment]
Yeah, I'll make one soon.

**Comment by TheColaber** [comment]
https://scratch.mit.edu/studios/30078251/

**Comment by TheColaber** [comment]
@FunctionalMetatable Do you prefer monk or mongoose?

**Comment by FunctionalMetatable** [comment]
> @FunctionalMetatable Do you prefer monk or mongoose? Edit: I think mongoose is better in our case, but I want to know what you think.

monk is much easier imo

**Comment by TheColaber** [comment]
Why is that? I think mongoose's schema thing makes it easier to organize our code more.

**Comment by TheColaber** [comment]
Also, do you think that this line:
https://github.com/FlagClicked/FlagClicked/blob/7ded56c5075b2e985c145bab51a42323aa722fd6/pages/login.vue#L42
Could be moved into the store? Particularly, right here?https://github.com/FlagClicked/FlagClicked/blob/7ded56c5075b2e985c145bab51a42323aa722fd6/store/auth.js#L13

**Comment by FunctionalMetatable** [comment]
> Also, do you think that this line:
> https://github.com/FlagClicked/FlagClicked/blob/7ded56c5075b2e985c145bab51a42323aa722fd6/pages/login.vue#L42
> 
> 
> Could be moved into the store? Particularly, right here?
> https://github.com/FlagClicked/FlagClicked/blob/7ded56c5075b2e985c145bab51a42323aa722fd6/store/auth.js#L13
> 
> 
> I think the client should try not to mess with cookies as much as possible.

thats the same thing

**Comment by FunctionalMetatable** [comment]
@TheColaber preview: https://flag-clicked-9gr.vercel.app/

**Comment by TheColaber** [comment]
Axios works for me...

**Comment by FunctionalMetatable** [comment]
> Axios works for me...

Doesnt work in production though

**Comment by Steve0Greatness** [comment]
@FunctionalMetatable it doesn't seem to work for me, but I might be doing something wrong.

**Comment by FunctionalMetatable** [comment]
@TheColaber could this be merged? I'll implement mongoose later

**Comment by TheColaber** [comment]
Okay, let me review some more...

**Comment by FunctionalMetatable** [comment]
> @TheColaber preview: [flag-clicked-9gr.vercel.app](https://flag-clicked-9gr.vercel.app/)
> 
> cc @CubeyTheCube @Steve0Greatness @markverb1

oops fixed

**Comment by FunctionalMetatable** [comment]
Everything gets really messed up w/ nuxt and same-site servers

**Comment by TheColaber** [comment]
Why is that?

**Comment by FunctionalMetatable** [comment]
> Why is that?

fetching itself sounds really strange

https://github.com/FlagClicked/FlagClicked/blob/d6e01d410aee96d26120221e19f90055c22da57c/store/auth.js#L27
bases are really annoying, server + client expansion is pretty hard

https://github.com/FlagClicked/FlagClicked/blob/d6e01d410aee96d26120221e19f90055c22da57c/store/index.js#L18
we have to use req.get host for fetching itself, we cannot communicate between server-middleware and stores unless fetching the same server again

**Comment by TheColaber** [comment]
> we have to use req.get host for fetching itself

you would have to do the same thing with a separate folder, and hold the whole URL in a variable. I wanted to keep it so you don't need to host 2 things at once. I'll try to tidy up the code as much as I can. I know nuxt auth exists but it has some problems.

**Comment by FunctionalMetatable** [comment]
> I wanted to keep it so you don't need to host 2 things at once.

there could be 2 folders, and there could be a bash/batch script for installing, running in development mode and a production script.

**Comment by FunctionalMetatable** [comment]
@TheColaber could we merge this today?
Also, when could the alpha/beta release go live?

**Comment by TheColaber** [comment]
I have some local changes on my computer and I won't be home for the next couple of days. I can merge this now if you really want. Beta can happen after I get home if you want.

**Comment by FunctionalMetatable** [comment]
i'll wait for your changes

**Comment by FunctionalMetatable** [comment]
most changes are good, refetching is not there anymore, nuxt directly gets data from monk (client-side fetches server)

File: store/auth.js [hunk 1]
```diff
@@ -18,61 +21,62 @@ export const mutations = {
...
-    }
-    var res;
+      me = await module.getSession(token);
+    } else {
+      var res;
+
+      try {
+        res = await fetch(`/auth/me`, {
+          credentials: "include"
+        });
```
Comment (TheColaber) [inline store/auth.js]: how did you get this to work?

Comment (FunctionalMetatable) [reply store/auth.js]: I stopped fetching the same server and used a plugin to connect to auth module (only on server). fetch is only used on client

**Comment by TheColaber** [comment]
This is very cool! Im so excited to merge this, and after that, merge nuxt into master. I have a couple backend changes but I had to leave my house :( Let's try to merge this today or tomorrow.

**Comment by FunctionalMetatable** [comment]
yessir lol

also:

TODO start work on tutorial page
Tutorial thumbnail uploading?

**Comment by TheColaber** [comment]
Css stuff etc, a bit more and we can merge nuxt into master

**Pull request merged**